lib/plug/conn.ex

```diff
@@ -446,13 +446,13 @@ defmodule Plug.Conn do
   `:sent`.
   """
   @spec put_resp_header(t, binary, binary) :: t
-  def put_resp_header(%Conn{resp_headers: headers, state: state} = conn, key, value) when
-      is_binary(key) and is_binary(value) and state != :sent do
-    %{conn | resp_headers: List.keystore(headers, key, 0, {key, value})}
+  def put_resp_header(%Conn{state: :sent}, _key, _value) do
```

Comment:
@josevalim it turned out that restriction to `@unsent` states is breaking `run_before_send` in `send_chunked` function since we're making state change there https://github.com/lexmag/plug/blob/sent-state/lib/plug/conn.ex#L359 (same for `send_file` function).